infra/src/main/kotlin/fi/fta/geoviite/infra/publication/PublicationService.kt

```diff
@@ -342,8 +342,13 @@ class PublicationService @Autowired constructor(
             rlId.trackNumberId
         }
         val trackNumbers = trackNumberService.getMany(DRAFT, referenceLineTrackNumberIds + requestIds.trackNumbers)
-        val revertTrackNumberIds = trackNumbers.filter(TrackLayoutTrackNumber::isDraft).map { it.id as IntId }
-        val draftOnlyTrackNumberIds = trackNumbers.filter(TrackLayoutTrackNumber::isNewDraft).map { it.id as IntId }
+        val revertTrackNumberIds = trackNumbers
+            .filter(TrackLayoutTrackNumber::isDraft)
+            .map { it.id as IntId }
+        // If revert breaks other draft row references, they should be reverted too
+        val draftOnlyTrackNumberIds = trackNumbers
+            .filter { tn -> tn.isDraft && tn.contextData.officialRowId == null }
```

Comment:
Nään itse ehkä sinällään edelleen tilausta tuollaiselle entisen kaltaiselle `isNewDraft`-funkkarille. Sehän ei ota kantaa branchiin, joten sen pitäisi (ainakin sikäli kun pähkäilen oikein) toimia sellaisenaan sekä mainissa että designeissa.

Reply:
Tämä logiikkahan pitää kokonaisuutena miettiä design-puolella uudelleen. Siinä on luultavasti (toivottavasti) paljon kyllä yhteistä mutta tällaisenaan tämä ei toimi ollenkaan designeille. Voi myös olla että noille löytyy yhteinen selkeä abstraktio "isX", jolla näitä hoitaa, mutta "new" voi tarkoittaa montaa eri asiaa ja nuo käyttöpaikat on syytä joka tapauksessa katsoa  ensin että onko ne oikeasti sama. Siksi otin tuon flagin nyt toistaiseksi pois. Teen nyt tilalle sen dailyssä mietityn uuden flagin ja otan käyttöön siellä missä se näyttäisi oikealta.

Reply:
Juu, tämä tapaus tulikin aika lailla tosiaan käsiteltyä dailyssä 👍